src/machine.rs

```diff
@@ -110,4 +114,45 @@ impl Machine {
         let machine_file = MachineFile::from(machine_file_response.data);
         Ok(machine_file)
     }
+
+    pub async fn ping(&self) -> Result<(), Error> {
+        let client: Client = Client::default();
+        let _response: Response<MachineResponse> = client
+            .post(&format!(
+                "machines/{}/actions/ping", self.id),
+                None::<&()>,
+                None::<&()>
+            )
+            .await?;
+        Ok(())
+    }
+
+
+    pub fn monitor(self: Arc<Self>, heartbeat_interval: Duration, tx: Option<Sender<Error>>) -> BoxFuture<'static, ()> {
+        
+        async move {
+            let mut interval_stream = futures::stream::unfold((), move |_| {
+                let delay = futures_timer::Delay::new(heartbeat_interval);
```

Comment:
Suppose we set the heartbeat interval according to the policy `heartbeatDuration`. In that case, it may not be sent on time due to network issues.

Referring to the code from https://github.com/keygen-sh/keygen-go/blob/master/machine.go#L152, I also recommend using the same way to subtract 30 seconds from the `heartbeat_interval.`

```rust
let delay = futures_timer::Delay::new(heartbeat_interval - Duration::from_secs(30));
```

Reply:
The heartbeat duration is a parameter, to the method and so I wanted the final user to be able to set the parameter to whatever they want. If we then subtracted 30s from it it could be surprising behavior. In go the duration is not a parameter as it is being read from the configuration, which we can not do as the config has an Option on the HeartbeatDuration. Which would mean we have to decide on a default value or throw an error when it is not available and I would prefer to leave that decision to the user of the library and their preferences.

Reply:
That makes sense, but I think we should reflect a bit on the example. 

Changing `let interval = Duration::from_secs(machine.heartbeat_duration.unwrap_or(570) as u64);` to `let interval = Duration::from_secs(machine.heartbeat_duration.unwrap_or(600) - 30 as u64);` would be better.